server/expression/record.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2025 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package expression
+
+import (
+	"github.com/cockroachdb/errors"
+	"github.com/dolthub/go-mysql-server/sql"
+	vitess "github.com/dolthub/vitess/go/vt/sqlparser"
+
+	pgtypes "github.com/dolthub/doltgresql/server/types"
+)
+
+// NewRecordExpr creates a new record expression.
+func NewRecordExpr() *RecordExpr {
+	// Initialize the record expression with the generic Record type. When the analyzer
+	// resolves the children of the InjectedExpr and sets them by calling
+	// WithResolvedChildren(), the type will be updated with the exact field types.
+	return &RecordExpr{
+		typ: pgtypes.Record,
+	}
+}
+
+// RecordExpr is a set of sql.Expressions wrapped together in a single value.
+type RecordExpr struct {
+	exprs []sql.Expression
+	typ   *pgtypes.DoltgresType
```

Comment:
If this always returns `[]pgtypes.RecordValue`, then wouldn't the type always be the same thing?